### ISSUE TITLE
Clarification regarding algorithm 1

### ISSUE THREAD

**Issue by kirk86** [prafullasd/caffe-cpu#1]
Hi thanks for this,
I was wondering if you could clarify a few things in regards to Algorithm 1 in Section 3?
For the sake of argument suppose the same setting of Section 4.1 in MNIST example for multi-class classification with unique labels (i.e. each class only one label).

1) In this setting how is `M` computed for MNIST?
2) What is `pred` in this case, logits of the model or softmax output?
3) `exp` which is the targets is it one-hot encoded or something else?
